ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/WithdrawalRequestEncoder.java

```diff
@@ -22,15 +23,36 @@
 
 public class WithdrawalRequestEncoder {
 
-  public static void encode(final WithdrawalRequest withdrawalRequest, final RLPOutput rlpOutput) {
+  /**
+   * Encodes a Request into RLP format if it is a WithdrawalRequest.
+   *
+   * @param request The Request to encode, which must be a WithdrawalRequest.
+   * @param rlpOutput The RLPOutput to write the encoded data to.
+   * @throws IllegalArgumentException if the provided request is not a WithdrawalRequest.
+   */
+  public static void encode(final Request request, final RLPOutput rlpOutput) {
+    if (!(request instanceof WithdrawalRequest withdrawalRequest)) {
```

Comment:
Rather use the `Request.getType()` and check it is a deposit rather than doing an instanceof